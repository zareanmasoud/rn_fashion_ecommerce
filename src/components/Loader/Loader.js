import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import styles from './styles';
// import PropTypes from 'prop-types';

export const Loader = props => {
  return (
    <View style={styles.loading}>
      <ActivityIndicator size={props.size}/>
    </View>
  );
};

Loader.defaultProps = {
  size: 'small'
};

// Loader.propTypes = {
//   size: PropTypes.oneOf(['small', 'large'])
// };

