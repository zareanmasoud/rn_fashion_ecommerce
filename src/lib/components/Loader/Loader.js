import React from 'react';
import {ActivityIndicator, View} from 'react-native';
import styles from './styles';

type Props = {
  size?: 'small' | 'large',
};

const Loader = ({size}: Props) => {
  return (
    <View style={styles.loading}>
      <ActivityIndicator size={size}/>
    </View>
  );
};

Loader.defaultProps = {
  size: 'small',
};

export default Loader;

