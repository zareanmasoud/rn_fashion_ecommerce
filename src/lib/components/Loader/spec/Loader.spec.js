import React from 'react';
import {shallow} from 'enzyme';
import Loader from '../Loader';

describe('Loader Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<Loader />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
