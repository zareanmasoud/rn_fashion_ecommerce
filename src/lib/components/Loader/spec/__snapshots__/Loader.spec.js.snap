// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "marginTop": 20,
      "position": "absolute",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
